### PR TITLE
Shopping: Detect WooCommerce status

### LINKED ISSUE #11383: Shopping: Detect WooCommerce status
## Feature Description

Add a check to see if the woocommerce plugin is installed. If not installed when show an error / warning message to user in the shopping settings in the dashboard. We will also need to check to see if the plugin is activated ( and setup correctly ). 

At the same time, we may want to define a minimum version that we support and show a message to user to upgrade. 

There is also code to see if woo is installed as pass this to config. 

https://github.com/GoogleForCreators/web-stories-wp/blob/782fd71d2f74752a62ea69fe1173f530b8f85664/includes/Tracking.php#L180-L182

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary
Add a check to see if the woocommerce plugin is installed. If not installed when show an error / warning message to user in the shopping settings in the dashboard. We will also need to check to see if the plugin is activated ( and setup correctly ).

## User-facing changes
Screenshot 2022-05-23 at 16 28 02
Screenshot 2022-05-23 at 16 26 12
Screenshot 2022-05-23 at 16 25 52

## Testing Instructions

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1. Enable shopping experiment. 
2. Go to dashboard > settings. 
3. Change shopping setting to woocommerce. 
4. See message saying to install woocommerce. 
5. Install woocommerce ( do not activate )
6. Go to dashboard > settings. 
7. See message saying to active woocommerce. 
8. Activate woocommerce. 
9. See manage message. 

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #11383

### CODE REVIEW

File: packages/wp-dashboard/src/components/editorSettings/shopping/index.js [hunk 1]
```diff
@@ -68,6 +83,22 @@ function Shopping({
     [updateSettings]
   );
 
+  const onWoocommerceClick = useCallback(
+    (evt) => trackClick(evt, 'click_site_kit_link'),
```
Comment (swissspidy) [inline packages/wp-dashboard/src/components/editorSettings/shopping/index.js]:
```suggestion
    (evt) => trackClick(evt, 'click_woocommerce_link'),
```

File: includes/Shopping/Woocommerce_Query.php [hunk 2]
```diff
@@ -42,9 +59,10 @@ class Woocommerce_Query implements Product_Query {
 	 * @return Product[]|WP_Error
 	 */
 	public function get_search( string $search_term ) {
+		$status = $this->woocommerce->get_plugin_status();
 
-		if ( ! function_exists( 'wc_get_products' ) ) {
-			return new WP_Error( 'rest_unknown', __( 'Woocommerce is not installed.', 'web-stories' ), [ 'status' => 400 ] );
+		if ( ! $status['active'] ) {
+			return new WP_Error( 'rest_woocommerce_not_installed', __( 'Woocommerce is not installed.', 'web-stories' ), [ 'status' => 400 ] );
```
Comment (swissspidy) [inline includes/Shopping/Woocommerce_Query.php]: While at it

```suggestion
			return new WP_Error( 'rest_woocommerce_not_installed', __( 'WooCommerce is not installed.', 'web-stories' ), [ 'status' => 400 ] );
```

File: includes/Integrations/Woocommerce.php [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+			}
+		} elseif ( $is_installed ) {
+			if ( current_user_can( 'activate_plugin', $plugin ) ) {
+				$link = admin_url( 'plugins.php' );
+			}
+		} elseif ( current_user_can( 'install_plugins' ) ) {
+			$link = admin_url(
+				add_query_arg(
+					[
+						's'   => rawurlencode( __( 'Woocommerce', 'web-stories' ) ),
```
Comment (swissspidy) [inline includes/Integrations/Woocommerce.php]:
```suggestion
						's'   => rawurlencode( __( 'WooCommerce', 'web-stories' ) ),
```

File: includes/Integrations/Woocommerce.php [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Web_Stories\Integrations;
+
+/**
+ * Class Woocommerce.
+ */
+class Woocommerce {
```
Comment (swissspidy) [inline includes/Integrations/Woocommerce.php]: Should we spell this `WooCommerce`?

File: includes/Integrations/Woocommerce.php [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+	 * @since 1.21.0
+	 *
+	 * @return array Plugin status.
+	 */
+	public function get_plugin_status(): array {
+		$plugin       = 'woocommerce/woocommerce.php';
+		$is_installed = \array_key_exists( $plugin, get_plugins() );
+		$is_active    = $this->is_plugin_active();
+		$can_manage   = false;
+		$link         = '';
```
Comment (swissspidy) [inline includes/Integrations/Woocommerce.php]: We could link to `__( 'https://wordpress.org/plugins/woocommerce/', 'web-stories' )` by default. Then we can use that link if WooCommerce is installed but the user cannot activate plugins (or if WooCommerce is not installed and user cannot install it)

Similar to how we do this for Site Kit

File: packages/wp-dashboard/src/components/editorSettings/shopping/index.js [hunk 6]
```diff
@@ -44,6 +46,18 @@ export const TEXT = {
...
   ),
+  WOOCOMMERCE_INSTALL: __(
+    'Install <a>WooCommerce</a> to easily add shoppable products to Web Stories.',
+    'web-stories'
+  ),
+  WOOCOMMERCE_ACTIVE: __(
+    'Activate <a>WooCommerce</a> to easily add shoppable products to Web Stories.',
+    'web-stories'
+  ),
+  WOOCOMMERCE_MANAGED: __(
```
Comment (swissspidy) [inline packages/wp-dashboard/src/components/editorSettings/shopping/index.js]: `MANAGED` -> `MANAGE`?

File: packages/wp-dashboard/src/components/editorSettings/shopping/test/shopping.js [hunk 7]
```diff
@@ -58,6 +66,72 @@ describe('Editor Settings: Shopping <Shopping />', function () {
     expect(sectionHeader).toBeInTheDocument();
   });
 
+  it('should render Woocmmerce inputs and helper texts', function () {
```
Comment (timarney) [inline packages/wp-dashboard/src/components/editorSettings/shopping/test/shopping.js]:
```suggestion
  it('should render Woocommerce inputs and helper texts', function () {
```

File: packages/wp-dashboard/src/components/editorSettings/shopping/test/shopping.js [hunk 8]
```diff
@@ -58,6 +66,72 @@ describe('Editor Settings: Shopping <Shopping />', function () {
...
+    );
+
+    const element = screen.getByTestId('woocommerceMessage');
+    expect(element).toBeInTheDocument();
+    expect(element).toHaveTextContent(
+      'Manage products in WooCommerce to make them available in Web Stories.'
+    );
+  });
+
+  it('should render Woocmmerce inputs and activation message', function () {
```
Comment (timarney) [inline packages/wp-dashboard/src/components/editorSettings/shopping/test/shopping.js]:
```suggestion
  it('should render Woocommerce inputs and activation message', function () {
```

File: packages/wp-dashboard/src/components/editorSettings/shopping/test/shopping.js [hunk 9]
```diff
@@ -58,6 +66,72 @@ describe('Editor Settings: Shopping <Shopping />', function () {
...
+    );
+
+    const element = screen.getByTestId('woocommerceMessage');
+    expect(element).toBeInTheDocument();
+    expect(element).toHaveTextContent(
+      'Activate WooCommerce to easily add shoppable products to Web Stories.'
+    );
+  });
+
+  it('should render Woocmmerce inputs and install message', function () {
```
Comment (timarney) [inline packages/wp-dashboard/src/components/editorSettings/shopping/test/shopping.js]:
```suggestion
  it('should render Woocommerce  inputs and install message', function () {
```

**Comment by timarney** [comment]
Tested locally -- working well 👍

File: includes/Integrations/WooCommerce.php [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+ * limitations under the License.
+ */
+
+namespace Google\Web_Stories\Integrations;
+
+/**
+ * Class WooCommerce.
+ */
+class WooCommerce {
+	protected const PLUGIN = 'woocommerce/woocommerce.php';
```
Comment (swissspidy) [inline includes/Integrations/WooCommerce.php:33]: Curious, why a class const for something that is only used in 1 method and could just be a variable there?

File: packages/wp-dashboard/src/components/editorSettings/shopping/test/shopping.js [hunk 11]
```diff
@@ -58,6 +66,72 @@ describe('Editor Settings: Shopping <Shopping />', function () {
...
+    );
+
+    const element = screen.getByTestId('woocommerceMessage');
+    expect(element).toBeInTheDocument();
+    expect(element).toHaveTextContent(
+      'Activate WooCommerce to easily add shoppable products to Web Stories.'
+    );
+  });
+
+  it('should render WooCommerce  inputs and install message', function () {
```
Comment (swissspidy) [inline packages/wp-dashboard/src/components/editorSettings/shopping/test/shopping.js:111]:
```suggestion
  it('should render WooCommerce inputs and install message', function () {
```

**Review by timarney — APPROVED** [review]

**Pull request merged**